### PR TITLE
Carefulclickall - Drop Down - Fixes #405

### LINKED ISSUE #405: Too easy to accidentally override existing marks with 'set status for all users' radiobutton.
We've found the "set status for all users" radiobuttons to be both essential and infuriating. It is normal for some students' statuses to be set to a state that indicates a "excused absence" in advance of a session, which means the "set status for all users" cannot be used to mark all those who are expected to be present as present at the time of the session.

We've implemented a fix so that the "set status for all users" will only set status for users whose status is not already set, unless shift-clicked, in which case it will set for all users and override existing statuses.

### PR DESCRIPTION
**Pull request by maksudr**
Building on the idea in #405. Converting shift-click functionality into a dropdown.

@danmarsden I have made the UX changes discussed. Let me know what you think!

### CODE REVIEW

**Comment by maksudr** [comment]
dropdown1
dropdown2

**Comment by danmarsden** [comment]
Sorry - I thought I'd responded to this last week! yeah - that's more like it I think, although the selected value should be stored as a user preference so that when viewing the page, the users preferred option is always selected.

 @nwp90 what do you think?

(obviously we'd need to adjust some css here, but I think it's in the right location and it feels more logical.)

**Comment by nwp90** [comment]
Doesn't feel 100%, but it'll do the job.

One thing that bugs me slightly is that it seems to me that the more destructive option should always be the one requiring extra effort to achieve.
 
I've not done much mobile stuff so I don't know whether there's a "standard" way of doing this kind of thing (yet?)...

File: renderer.php [hunk 1]
```diff
@@ -649,7 +652,21 @@ protected function render_attendance_take_list(attendance_take_data $takedata) {
         foreach ($extrasearchfields as $field) {
             $row->cells[] = '';
         }
-        $row->cells[] = html_writer::div(get_string('setallstatuses', 'attendance'), 'setallstatuses');
+
+        if (get_user_preferences('mod_attendance_statusdropdown') == 'unselected') {
```
Comment (danmarsden) [inline renderer.php]: might be good to drop this into a helper function as it's used twice in the code - we should also be using lang strings for the "all" and "unselected" items shouldn't we?

**Comment by maksudr** [comment]
@danmarsden I'verefactored the code with a helper function. Should be good to go :+1:

**Comment by danmarsden** [comment]
merged this in manually! - thanks guys!! :-)

**Pull request closed without merging**